packages/store/types/tests/action.lint.ts

```diff
@@ -390,3 +394,68 @@ describe('[TEST]: Action Operator Types', () => {
     });
   });
 });
+
+describe('[TEST]: Action Decorator Types', () => {
+  class ActionString {
+    static type = 'ACTION 1';
+    constructor(public a: string) {}
+  }
+
+  class ActionNumber {
+    static type = 'ACTION 2';
+    constructor(public c: number) {}
+  }
+
+  class ActionString2 {
+    static type = 'ACTION 3';
+    constructor(public a: string) {}
+  }
```

Comment:
What this PR introduces is so much better than before, but if the method parameter has the class with the `static type` information then we could provide an even stricter matching.
Is it possible to also check if the correct action type is provided, not just the payload?
So it would be an error if we have:
```typescript
@Action(ActionString)
myAction(ctx: StateContext<any>, action: ActionString2) {
  // ....
}
```

Obviously if the parameter is specified with only the payload shape then we will have to fall back to the payload matching.

Note: We could do this even stricter typing as another PR after this.